xdsl/dialects/memref_stream.py

```diff
@@ -61,6 +66,71 @@
 from xdsl.utils.hints import isa
 from xdsl.utils.str_enum import StrEnum
 
+_StreamTypeElement = TypeVar("_StreamTypeElement", bound=Attribute, covariant=True)
+_StreamTypeElementConstrT = TypeVar("_StreamTypeElementConstrT", bound=Attribute)
+
+
+@irdl_attr_definition
+class ReadableStreamType(
```

Comment:
Isn't that how this started 5 or 6 PRs ago? (admittedly with a much more complicated stream type)

Reply:
the type is very simple atm, so I'm just wondering if there is an immediate design decision coming up, hence the split

Reply:
I thought of this for a bit, but then I thought that with the current setup it would only save about 6 lines per class definition, namely the name, init, and `get_element_type`, as we don't have a working setup for making the constraint a classmethod. Locally having it in a utils felt a bit silly, but it's very subjective, happy to change if you feel like it would be better.